internal/infrastructure/kubernetes/proxy_deployment.go

```diff
@@ -91,43 +91,19 @@ func (i *Infra) expectedProxyDeployment(infra *ir.Infra) (*appsv1.Deployment, er
 					SecurityContext:               deploymentConfig.Pod.SecurityContext,
 					Affinity:                      deploymentConfig.Pod.Affinity,
 					Tolerations:                   deploymentConfig.Pod.Tolerations,
-					Volumes: []corev1.Volume{
-						{
-							Name: "certs",
-							VolumeSource: corev1.VolumeSource{
-								Secret: &corev1.SecretVolumeSource{
-									SecretName: "envoy",
-								},
-							},
-						},
-						{
-							Name: "sds",
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{
-										Name: expectedResourceHashedName(infra.Proxy.Name),
-									},
-									Items: []corev1.KeyToPath{
-										{
-											Key:  sdsCAFilename,
-											Path: sdsCAFilename,
-										},
-										{
-											Key:  sdsCertFilename,
-											Path: sdsCertFilename,
-										},
-									},
-									DefaultMode: pointer.Int32(int32(420)),
-									Optional:    pointer.Bool(false),
-								},
-							},
-						},
-					},
+					Volumes:                       expectedDeploymentVolumes(infra.Proxy.Name, deploymentConfig),
 				},
 			},
 		},
 	}
 
+	// apply merge patch to deployment
+	if merged, err := deploymentConfig.ApplyMergePatch(deployment); err == nil {
+		deployment = merged
+	} else {
+		return nil, fmt.Errorf("failed to apply merge patch to deployment: %w", err)
+	}
```

Comment:
```suggestion
	// apply merge patch to deployment
	merged, err := deploymentConfig.ApplyMergePatch(deployment)
	if err != nil {
		return nil, fmt.Errorf("failed to apply merge patch to deployment: %w", err)
	}
	return merged, nil
```
That said, I recommend patching it in-place in ApplyMergePatch, and then doing:
```
	// apply merge patch to deployment, 
	if err := deploymentConfig.ApplyMergePatch(deployment); err != nil {
		return nil, fmt.Errorf("failed to apply merge patch to deployment: %w", err)
	}
	return deployment, nil
```
Since you'll never need the original. This is simpler and might avoid bugs in the future related to returning the same pointer when there is no patch.

Reply:
I applied the flatter logic in your suggestion. I'm not going to dip into changing the function signature of the backported code since we're hopefully not keeping our fork alive indefinitely.